buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1982,17 +1982,25 @@ class BeamModulePlugin implements Plugin<Project> {
         return argList.join(' ')
       }
 
-      project.ext.toxTask = { name, tox_env ->
+      project.ext.toxTask = { name, tox_env, needsExpansionServiceJar = false ->
+        project.evaluationDependsOn(":sdks:java:io:expansion-service")
         project.tasks.create(name) {
           dependsOn 'setupVirtualenv'
           dependsOn ':sdks:python:sdist'
+          if (needsExpansionServiceJar) {
+            dependsOn ':sdks:java:io:expansion-service:shadowJar'
+          }
 
           doLast {
             // Python source directory is also tox execution workspace, We want
             // to isolate them per tox suite to avoid conflict when running
             // multiple tox suites in parallel.
             project.copy { from project.pythonSdkDeps; into copiedSrcRoot }
-
+            if (needsExpansionServiceJar) {
+              def expansionServiceJar =  project.project(':sdks:java:io:expansion-service').shadowJar.archivePath
+              def expansionServiceDestinationDir = "${copiedSrcRoot}/sdks/java/io/expansion-service/build/libs"
```

Comment:
Yeah, these are python tox suites that include pythonLint, pythonFormatter, pythonDocs that do not require expansion service to be built for their executions. Only test suites require them.

Reply:
Sorry, I didn't read carefully. Tox suites use separate environment and BeamJarExpansionService('sdks:java:io:expansion-service:shadowJar') cannot find the jar if it is not copied into the tox's root path

Reply:
Why to we need a jar in tox (unit test) suite?

Reply:
I understand that we may need it for postcommit (integration tests), but why precommits-on-tox?

Reply:
Because I understood that we want these tests to run in precommit with fn_api runner. If we don't then I'll leave them running in postcommit suites (which is good enough as I assume) and disable them in precommit.